cli/src/commands/version.rs

```diff
@@ -22,14 +22,66 @@ use crate::ui::Ui;
 
 /// Display version information
 #[derive(clap::Args, Clone, Debug)]
-pub(crate) struct VersionArgs {}
+pub(crate) struct VersionArgs {
+    /// Display only the version number and nothing else.
+    #[arg(long)]
+    pub(crate) numeric_only: bool,
+
+    /// Display build information.
+    #[arg(long, conflicts_with = "numeric_only")]
+    pub(crate) build_info: bool,
+}
 
 #[instrument(skip_all)]
 pub(crate) fn cmd_version(
     ui: &mut Ui,
     command: &CommandHelper,
-    _args: &VersionArgs,
+    args: &VersionArgs,
 ) -> Result<(), CommandError> {
-    write!(ui.stdout(), "{}", command.app().render_version())?;
+    let base_version = command.app().get_version().unwrap();
+    let (version, git_commit) = if let Some(git_commit) = option_env!("JJ_GIT_COMMIT") {
+        // in a release build, don't include the commit hash in the
+        // `--numeric-version` or normal output. GH-3629
+        if option_env!("JJ_RELEASE_BUILD").is_some() {
+            (String::from(base_version), git_commit)
+        } else {
+            let short_commit = &git_commit[..12];
+            (format!("{}-{}", base_version, short_commit), git_commit)
+        }
+    } else {
+        (String::from(base_version), "unknown")
+    };
+
+    if args.numeric_only {
+        writeln!(ui.stdout(), "{}", version)?;
+        return Ok(());
+    }
+
+    write!(
+        ui.stdout(),
+        "\
+Jujutsu version control system; jj {}
+Copyright (C) 2019-2024 The Jujutsu Authors
+License: \
+         Apache License, Version 2.0
+There is NO WARRANTY, to the extent permitted by law.
+
+Homepage: \
+         <https://martinvonz.github.io/jj>
+Report bugs: <https://github.com/martinvonz/jj/issues>
```

Comment:
I was thinking: when people report bugs, we'd like them to include `jj version --build-info`.

So, perhaps we could have `jj version` print two lines:

```
Jujutsu version control system; jj {}
For more details, run: jj version --details
```

and then have `jj version --details` print the build info, homepage, etc. (The license info could go there too, if you insist, though as I said elsewhere, I'd prefer to just omit it)

BTW, it's a bit of a hack, but we could have `jj version -v` and `jj version --verbose` to act like `jj version --details`.

Reply:
Good idea, I've adopted this format. The `--verbose` flag seems unneeded, at least.